src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private Entry<K, V>[] buckets;
+    private int size;
+    private int capacity;
+    private int resizeCount;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public MyHashMap(int initialCapacity) {
+        capacity = initialCapacity;
+        buckets = new Entry[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Entry<K, V> entry = buckets[index];
+        while (entry != null) {
+            if (key == entry.key || key != null && key.equals(entry.key)) {
+                entry.value = value;
+                return;
+            }
+            entry = entry.next;
+        }
+        Entry<K,V> newEntry = new Entry<>(key, value);
+        newEntry.next = buckets[index];
+        buckets[index] = newEntry;
+        size++;
+        if (size > getThreshold()) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Entry<K, V> entry = buckets[index];
+
+        while (entry != null) {
+            if (key == entry.key || key != null && key.equals(entry.key)) {
+                return entry.value;
+            }
+            entry = entry.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private double getThreshold() {
+        return LOAD_FACTOR * (double) DEFAULT_CAPACITY * Math.pow(RESIZE_FACTOR, resizeCount);
+    }
+
+    private void resize() {
+        int newCapacity = buckets.length * RESIZE_FACTOR;
+        Entry<K, V>[] newBuckets = new Entry[newCapacity];
+
+        for (int i = 0; i < buckets.length; i++) {
```

Comment:
see the example
![image](https://github.com/mate-academy/jv-hash-map/assets/20326858/9549418a-477e-4a8d-95ed-153bfc50ce91)